sympy/matrices/matrices.py

```diff
@@ -31,6 +31,33 @@ def _iszero(x):
     """Returns True if x is zero."""
     return x.is_zero
 
+def _next_pivot_down_right(matrix, pivot_row, pivot_col, iszerofunc):
+    """Locates the pivot position by testing if the element at (pivot_row, pivot_col) is zero,
+    and then incrementing the row index until a non-zero element is encountered, or the row index is out of range.
+    If a pivot is not found, then the column index is incremented, the row index is reset to the original value,
+    and the search is repeated.
+    Returns the row and column indices of the pivot position if a pivot is encountered, otherwise returns the
+    dimensions of the input matrix."""
+    for candidate_pivot_col in range(pivot_col, matrix.cols):
+        for candidate_pivot_row in range(pivot_row, matrix.rows):
+            if not iszerofunc(matrix[candidate_pivot_row, candidate_pivot_col]):
```

Comment:
If `iszerfunc` uses `True`, `False`, `None` logic, you may want to write this line as

```
if iszerofunc(matrix[candidate_pivot_row, candidate_pivot_col]) == False:
```

Look at https://github.com/sympy/sympy/blob/master/sympy/matrices/matrices.py#L5077 for a heuristic for finding a pivot.